### PR TITLE
feat: Add Parameters Utility

### PR DESCRIPTION
**Pull request by amirkaws**
**Issue number:** #160

### Changes

This PR adds parameters utility to Lambda Powertools for features parity across Powertools languages. it includes supports for [AWS Systems Manager Parameter Store](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html), [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/), [AWS AppConfig](https://docs.aws.amazon.com/appconfig/latest/userguide/what-is-appconfig.html), [Amazon DynamoDB](https://aws.amazon.com/dynamodb/), or bring your own.

### User experience

1- Install the package

```
dotnet nuget add AWS.Lambda.Powertools.Parameters
```
2- Grants relevant IAM permissions
3- Configure and use relevant parameter provider for example

```c#
using AWS.Lambda.Powertools.Parameters;
using AWS.Lambda.Powertools.Parameters.SimpleSystemsManagement;

public class Function
{
    public async Task<APIGatewayProxyResponse> FunctionHandler
        (APIGatewayProxyRequest apigProxyEvent, ILambdaContext context)
    {
        // Get SSM Provider instance
        ISsmProvider ssmProvider = ParametersManager.SsmProvider;
            
        // Retrieve a single parameter
        string? value = await ssmProvider
            .GetAsync("/my/parameter")
            .ConfigureAwait(false);
        
        // Retrieve multiple parameters from a path prefix
        // This returns a Dictionary with the parameter name as key
        IDictionary<string, string?> values = await ssmProvider
            .GetMultipleAsync("/my/path/prefix")
            .ConfigureAwait(false);
    }
}
```
Refer to [Parameters read me](https://github.com/amirkaws/aws-lambda-powertools-dotnet/blob/amirkaws-feature-parameters/docs/utilities/parameters.md) for more information.

## Checklist

Please leave checklist items unchecked if they do not apply to your change.

* [x] [Meets tenets criteria](https://awslabs.github.io/aws-lambda-powertools-dotnet/tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-dotnet/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

No

**RFC issue number**: #160

Checklist:

* [x] Migration process documented
* [x] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

-----
[View rendered docs/utilities/parameters.md](https://github.com/amirkaws/aws-lambda-powertools-dotnet/blob/amirkaws-feature-parameters/docs/utilities/parameters.md)

### CODE REVIEW

File: libraries/src/AWS.Lambda.Powertools.Parameters/AWS.Lambda.Powertools.Parameters.csproj [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>net6.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <PackageReference Include="AWSSDK.AppConfigData" Version="3.7.0.110" />
```
Comment (hjgraca) [inline libraries/src/AWS.Lambda.Powertools.Parameters/AWS.Lambda.Powertools.Parameters.csproj:10]: Is this still needed, now that we don't support appconfig?

File: libraries/src/AWS.Lambda.Powertools.Parameters/AWS.Lambda.Powertools.Parameters.csproj [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      <PackageReference Include="AWSSDK.DynamoDBv2" Version="3.7.5.10" />
```
Comment (hjgraca) [inline libraries/src/AWS.Lambda.Powertools.Parameters/AWS.Lambda.Powertools.Parameters.csproj:11]: Can you update to the latest versions of these packages.
AWSSDK.DynamoDBv2 -> 3.7.103.1
AWSSDK.SecretsManager -> 3.7.102.38
AWSSDK.SimpleSystemsManagement -> 3.7.104.29

**Review by hjgraca — CHANGES_REQUESTED** [review]
Can you please implement the `PowertoolsConfiguration SetExecutionEnvironment`. To add Parameters to the user agent. Could be on the singleton constructor of the parameters. And add the test. Have a look [here](https://github.com/awslabs/aws-lambda-powertools-dotnet/blob/develop/libraries/src/AWS.Lambda.Powertools.Tracing/Internal/XRayRecorder.cs#L50) and [here](https://github.com/awslabs/aws-lambda-powertools-dotnet/blob/develop/libraries/tests/AWS.Lambda.Powertools.Tracing.Tests/XRayRecorderTests.cs#L14)

**Review by hjgraca — APPROVED** [review]
GTM

**Comment by aviv-akhansari** [comment]
I'm wondering why the AppConfig feature was removed?

**Comment by hjgraca** [comment]
@aviv-akhansari thanks for your question. We removed the feature from this pull request because there were some API changes on AppConfig that we could not tackle in time to get this release out. 
We were also waiting on customer feedback to get it back, and now we have it 😄  and will begin the work to add AppConfig to Powertools.

Again thanks for using Powertools and we will keep you posted on the progress.

**Pull request merged**